scripts/build/set-azp-variables.ps1

```diff
@@ -0,0 +1,8 @@
+# Read variables shared between stages and tasks
+$Dir = "$env:AGENT_BUILDDIRECTORY\Azp-Variables"
+
+$Version = Get-Content "${Dir}\SONAR_PROJECT_VERSION"
```

Comment:
you are using `SONAR_PROJECT_VERSION` both in this storage and in `##vso[task.setvariable variable=SONAR_PROJECT_VERSION;]$Version` which from what I understand are different...

could you prefix the global one with `GLOBAL_SONAR_PROJECT_VERSION` to make it clear which is the one stored globally?

(or maybe I don't understand something and we need to sync offline to better understand)

Reply:
Environment variable `SONAR_PROJECT_VERSION` (line 8) is holding the value of sonar project version for current job.
This value is persisted in `Azp-Variables\SONAR_PROJECT_VERSION` file between stages and jobs.

It's important to have exactly the same name for both.

Reply:
ok, I didn't know `Azp-Variables\SONAR_PROJECT_VERSION` was a file